File: test_regress/t/t_savable_open_bad.pl
#!/usr/bin/env perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2003 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

scenarios(vlt => 1);

top_filename("t/t_savable.v");

compile(
    v_flags2 => ["--savable"],
    save_time => 500,
    );

execute(
    all_run_flags => ['+save_restore=1'],
    fails => 1,
    expect_filename => $Self->{golden_filename},
    );

ok(1);
1;

File: test_regress/t/t_savable_open_bad.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('vlt')
test.top_filename = "t/t_savable.v"

test.compile(v_flags2=["--savable"], save_time=500)

test.execute(all_run_flags=['+save_restore=1'], fails=True, expect_filename=test.golden_filename)

test.passes()
